canonical-web-and-design/maas-ui
# feat(ui): improve and align styling of machine commissioning+tests tabs

## Done

- Replaced `MachineCommissioningTable` with `MachineTestsTable`, which has been updated to drop the suppress column if there are no testing scripts
- Created a `ScriptStatus` component which just lumps together a script result's status icon and name
- Adjusted the styling to be more responsive

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the testing tab of a machine and adjust the viewport width
- Check that the column content adjusts according to the viewport width and looks ok on all screen sizes
- Check that the suppress checkbox shows for all results, but is disabled with a tooltip if the test did not fail
- Click the "View previous tests" action and check that the test history is aligned properly and icons are shown for all the results
- Click the "View details" actions and check that it takes you to the right place
- Go to the commissioning tab of a machine and do the same (except for the suppress checkbox)

## Fixes

Fixes #2144

## Screenshot

![2021-02-22_17-44](https://user-images.githubusercontent.com/25733845/108677578-98ab1f00-7535-11eb-92dd-e532ff6e4349.png)
